miner/worker.go

```diff
@@ -395,13 +435,32 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 				signedRecent, err := p.SignRecently(w.chain, head.Block.Header())
 				if err != nil {
 					log.Info("Not allowed to propose block", "err", err)
+					//invoke pre-mining by notify txpoolsnapshotloop without blocking if not available
```

Comment:
Maybe TxPoolPrefetch is not necessary for nodes which are not allowed to propose block.
TxPoolPrefetch is for validators to reduce the time of block import, so it can have more time to mine when it is inturn.